### PR TITLE
Add supportedThreadFeatures, threadVersion and SupportedWiFiBands attributes to some reference apps

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
Update the zap file of some reference apps to enable 
- supportedThreadFeatures
- threadVersion
- SupportedWiFiBands 

attributes of the network commissioning cluster
The affected apps are :
- all-clusters-app
- all-clusters-minimal-app
- ~~lock-app~~
- ~~lighting-app~~
- ~~lit-icd-server-app~~

**NOTE this enables both Wifi and Thread specific clusters for validation purposes regardless of the platform network technology. This should not be the normal configuration of a product and that is why it is not applied to all sample apps - _Open to better suggestion about_**

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Comment by andy31415** [comment]
@jmartinez-silabs the cirque failure may be a real failure. I see:

```
Using PASE connection
2024-01-10 05:57:01,495 [PythonMatterControllerTEST] ERROR 'ValueDecodeFailure' object has no attribute 'networks'
Traceback (most recent call last):
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 183, in <module>
    main()
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 170, in main
    raise ex
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 148, in main
    asyncio.run(flow.commission(commissioning.Parameters(
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/example_python_commissioning_flow.py", line 49, in commission
    await self.network_commissioning(parameter=parameter, node_id=node_id)
  File "/usr/local/lib/python3.8/dist-packages/chip/commissioning/commissioning_flow_blocks.py", line 217, in network_commissioning
    if network_commissioning_cluster_state.networks:
AttributeError: 'ValueDecodeFailure' object has no attribute 'networks'
ERROR:PythonMatterControllerTEST:'ValueDecodeFailure' object has no attribute 'networks'
Traceback (most recent call last):
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 183, in <module>
    main()
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 170, in main
    raise ex
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/python_commissioning_flow_test.py", line 148, in main
    asyncio.run(flow.commission(commissioning.Parameters(
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/runner/work/connectedhomeip/connectedhomeip/src/controller/python/test/test_scripts/example_python_commissioning_flow.py", line 49, in commission
    await self.network_commissioning(parameter=parameter, node_id=node_id)
  File "/usr/local/lib/python3.8/dist-packages/chip/commissioning/commissioning_flow_blocks.py", line 217, in network_commissioning
    if network_commissioning_cluster_state.networks:
AttributeError: 'ValueDecodeFailure' object has no attribute 'networks'
```

**Comment by jmartinez-silabs** [comment]
@andy31415 yes it seems like a real error.  I will try it tomorrow and fix it. I didn't get time to get back to it today

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
After discussing with @tcarmelveilleux, I reverted the config changes to the sample apps other than all-cluster-apps for now as we want to avoid users copying non-spec compliant configs from sample apps.

We are looking into ways to ensure that the attribute list matches the device network type in a compliant way. In combination with a valid feature-amp attribute, compliance tests should be able to validate devices.

File: src/app/clusters/network-commissioning/network-commissioning.cpp [hunk 1]
```diff
@@ -269,33 +266,61 @@ CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValu
...
-    case Attributes::SupportedWiFiBands::Id:
-        VerifyOrReturnError(mFeatureFlags.Has(Feature::kWiFiNetworkInterface), CHIP_NO_ERROR);
-        VerifyOrReturnError(mpDriver.Valid(), CHIP_NO_ERROR);
-
-        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
-            WiFiBand bandsBuffer[kMaxWiFiBands];
-            Span<WiFiBand> bands(bandsBuffer);
-            ReturnErrorOnFailure(mpDriver.Get<WiFiDriver *>()->GetSupportedWiFiBands(bands));
-
+    case Attributes::SupportedWiFiBands::Id: {
```
Comment (tcarmelveilleux) [inline src/app/clusters/network-commissioning/network-commissioning.cpp:269]: Let's wait for https://github.com/project-chip/connectedhomeip/pull/31349 to land. Will fix this after.

**Pull request merged**